website/content/docs/platform/k8s/vso/sources/vault.mdx

```diff
@@ -0,0 +1,283 @@
+Supported secrets engines: [kv-v2](/vault/docs/secrets/kv/kv-v2), [kv-v1](/vault/docs/secrets/kv/kv-v1)
+
+##### KV version 1 secret example
+
+The KV secrets engine's `kvv1` mount path is specified under `spec.mount` of `VaultStaticSecret` custom resource. Please consult [KV Secrets Engine - Version 1 - Setup](/vault/docs/secrets/kv/kv-v1#setup) for configuring KV secrets engine version 1. The following results in a request to `http://127.0.0.1:8200/v1/kvv1/eng/apikey/google` to retrieve the secret.
+
+```yaml
+---
+apiVersion: secrets.hashicorp.com/v1beta1
+kind: VaultStaticSecret
+metadata:
+  namespace: vso-example
+  name: vault-static-secret-v1
+spec:
+  vaultAuthRef: vault-auth
+  mount: kvv1
+  type: kv-v1
+  path: eng/apikey/google
+  refreshAfter: 60s
+  destination:
+    create: true
+    name: static-secret1
+```
+
+##### KV version 2 secret example
+
+The KV secrets engine's `kvv2` mount path is specified under `spec.mount` of `VaultStaticSecret` custom resource. Please consult [KV Secrets Engine - Version 2 - Setup](/vault/docs/secrets/kv/kv-v2#setup) for configuring KV secrets engine version 2. The following results in a request to `http://127.0.0.1:8200/v1/kvv2/data/eng/apikey/google?version=2` to retrieve the secret.
+
+```yaml
+---
+apiVersion: secrets.hashicorp.com/v1beta1
+kind: VaultStaticSecret
+metadata:
+  namespace: vso-example
+  name: vault-static-secret-v2
+spec:
+  vaultAuthRef: vault-auth
+  mount: kvv2
+  type: kv-v2
+  path: eng/apikey/google
+  version: 2
+  refreshAfter: 60s
+  destination:
+    create: true
+    name: static-secret2
+```
+
+### VaultPKISecret custom resource
+Provides the configuration necessary for the Operator to synchronize a single Vault *PKI* Secret to a single Kubernetes Secret.<br />
+Supported secrets engines: [pki](/vault/docs/secrets/pki)
+
+The PKI secrets engine's mount path is specified under `spec.mount` of `VaultPKISecret` custom resource. Please consult [PKI Secrets Engine - Setup and Usage](/vault/docs/secrets/pki/setup) for configuring PKI secrets engine. The following results in a request to `http://127.0.0.1:8200/v1/pki/issue/default` to generate TLS certificates.
+
+```yaml
+---
+apiVersion: secrets.hashicorp.com/v1beta1
+kind: VaultPKISecret
+metadata:
+  namespace: vso-example
+  name: vault-pki
+spec:
+  vaultAuthRef: vault-auth
+  mount: pki
+  role: default
+  commonName: example.com
+  format: pem
+  expiryOffset: 1s
+  ttl: 60s
+  namespace: tenant-1
+  destination:
+    create: true
+    name: pki1
+```
+
+### VaultDynamicSecret custom resource
+
+Provides the configuration necessary for the Operator to synchronize a single Vault *dynamic* Secret to a single Kubernetes Secret.<br />
+Supported secrets engines *non-exhaustive*: [databases](/vault/docs/secrets/databases), [aws](/vault/docs/secrets/aws),
+[azure](/vault/docs/secrets/azure), [gcp](/vault/docs/secrets/gcp), ...
```

Comment:
```suggestion
VaultDynamicSecret configures the secret operator to synchronize a single Vault
dynamic Secret to a single Kubernetes Secret.

Supported secrets engines:

- [databases](/vault/docs/secrets/databases)
- [aws](/vault/docs/secrets/aws)
- [azure](/vault/docs/secrets/azure)
- [gcp](/vault/docs/secrets/gcp)
```

Reply:
Adding "non-exhaustive" isn't helpful for readers. We should either say "all Vault dynamic secrets engines" or take the time to actually list the explicit engines that are currently supported.

Reply:
Let's save this for another review since these are related to the refactoring of the current docs.